iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -84,9 +107,11 @@ Viewer.prototype._preProcessiXBRL = function(n, inHidden) {
         $(n).wrap(wrapper);
         node = $(n).parent();
     }
-    node.addClass("ixbrl-element").data('ivid',n.getAttribute("id"));
+    var id = n.getAttribute("id");
+    node.addClass("ixbrl-element").data('ivid',id);
+    this._factsByDocument[id] = doc;
```

Comment:
This seems confusingly named.  Rename to `_documentsByFactId`?

Well, it took me quite some time to understand that `doc` is actually an index, not a reference to a jQuery/DOM object.  Perhaps `rename `doc` to `docIndex` throughout and possibly name `_docIndexesByFactId`?  Alternatively, perhaps some comments would help.

Reply:
Yes - that's much clearer.  Done in 0cb3e95.